ParProcCo/job_scheduler.py

```diff
@@ -0,0 +1,274 @@
+import logging
+import os
+import stat
+from pathlib import Path
+from datetime import timedelta
+from datetime import datetime
+
+import drmaa2 as drmaa2
+from drmaa2 import JobSession, JobTemplate, Drmaa2Exception
+
+
+class TemporarilyLogToFile:
+
+    def __init__(self, logger, path=None):
+        self.logger = logger
+        self.log_path = path
+        if self.log_path is not None:
+            self.file_handler = logging.FileHandler(self.log_path, "a+")
+            self.logger.addHandler(self.file_handler)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc):
+        self.close()
+
+    def close(self):
+        if self.log_path is not None:
+            self.file_handler.close()
+            self.logger.removeHandler(self.file_handler)
+            with open(self.log_path, "a+") as fp:
+                fp.write("\n")
+
+
+class JobScheduler:
+
+    def __init__(self, working_directory, cluster_output_dir, project, priority, cpus=16, timeout=timedelta(hours=2)):
+        """
+        JobScheduler can be used for cluster job submissions
+        Args:
+            working_directory (pathlib.Path or str): working directory
+            cluster_output_dir (pathlib.Path or str): cluster output directory
+            project (str): project name
+            priority (str): name of queue to submit to
+            cpus (int, optional): Number of CPUs to request. Defaults to 16.
+            timeout (int, optional): Timeout for cluster jobs in minutes. Defaults to 180 (2 hours).
+        """
+        self.working_directory = Path(working_directory)
+        self.cluster_output_dir = Path(cluster_output_dir)
+        self.project = self.check_project_list(project)
+        self.priority = self.check_queue_list(priority)
+        self.cpus = cpus
+        self.timeout = timeout
+        self.logger = logging.getLogger()
```

Comment:
Unnecessary as you do not manipulate the logging in this class. The `logging.error`, etc, calls are package level convenience functions that delegate to the default logger.